pkg/kubelet/container/runtime.go

```diff
@@ -305,6 +306,18 @@ type PodStatus struct {
 	SandboxStatuses []*runtimeapi.PodSandboxStatus
 }
 
+// ContainerResources represents the Resources allocated to the running container.
```

Comment:
Why do we need a new type here? We have more defined container resources that are CRI compatible and not pinned to cgroupsv1 here: https://github.com/kubernetes/kubernetes/blob/6ac2d8edc8606ab387924b8b865b4a69630080e0/staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.proto#L48-L54

Reply:
Upon looking closely, there is no conflict. The one above is v1.ContainerResources and what I have introduced in [KEP 2273](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2273-kubelet-container-resources-cri-api-changes) is runtimeapi.ContainerResources that at this point represents LinuxContainerResources or WindowsContainerResources according to the runtime platform and exists on the runtime-facing side of kubelet. (This is how it should have been done when CRI was written and in particular when UpdateContainerResources API was developed)